src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,63 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.util.Scanner;
+
+/**
+ * Represents a storage attached to the Duke bot.
+ */
+public class Storage {
+    private String filePath = "";
+
+    /**
+     * Constructor for the storage.
+     *
+     * @param filePath Path of the file that contains saved information from the last run.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    //@@author chengda300
+    //Reused from https://nus-cs2103-ay2223s1.github.io/website/schedule/week3/topics.html
+    // with minor modifications
+
+    /**
+     * Loads data from the file to restore the last run status.
+     *
+     * @param d Duke bot that is running.
+     * @param filePath Path of the file that contains saved information from the last run.
+     * @throws FileNotFoundException If file is not found.
+     */
+    public void loadData(Duke d, String filePath) throws FileNotFoundException {
+        File f = new File(filePath);
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            d.parse(s.nextLine());
```

Comment:
It seems your approach to loading back the data is to save the entire command strings and parse them when you want to load the data in, reconstructing the tasks this way. I think this could potentially result in save file bloating if there are multiple repeated delete commands, although this is not a big issue given the scale of the project. Would it be better to convert Tasks into some kind of save string format that you can use to reconstruct the Task?

Reply:
Yes, it is a lazy way of storing and loading data for small scale projects.
For large scale projects it should undo the action as if the previous action did not happen at all to keep the list as short as possible.